Alert: Issues in BookshelfSection Component – Need Help Resolving State Update & Key Prop Warnings

tanya-newStart opened the pull request:
What We Need Help With:

State Update Issue: Need help in reviewing the component to move any state updates or side-effects out of the render function to prevent re-renders during the rendering of the BookshelfSection.
Key Prop Issue: Need help making that each child rendered within the .map() has a unique key prop based on the book’s identifier which is actually unique according to database set up

Files to Review:
BookshelfSection.jsx: Contains the problematic code that triggers both warnings.

tanya-newStart committed:
Update package-lock.json after reinstalling dependencies

tanya-newStart committed:
Merge remote-tracking branch 'upstream/main'

tanya-newStart committed:
Initial set up of env files and npm i

tanya-newStart committed:
Create goalsController to handle endpoitns, goalsRouter, and buildGoalDto

tanya-newStart committed:
all endpoints tested in Postman

tanya-newStart committed:
Create GoalsWidget component

tanya-newStart committed:
Add handleUpdateGoal and handleRefreshGoal functionality to GoalsWidget

tanya-newStart committed:
Get rid of 'reset' status

tanya-newStart committed:
Restore Bookshelf and related components from upstream/main

tanya-newStart committed:
Get rid of strict mode

magdazelena commented on app/app/components/Profile.jsx:
```diff
@@ -1,28 +1,30 @@
+    wishToRead: [],
+  });
 
   useEffect(() => {
```
first of all, I would suggest to spread the fetching data into a separate effect each. It is easy to get things bloated and they are independent from each other anyway.

tanya-newStart committed:
Create bookCount context

magdazelena commented on app/app/components/Profile.jsx:
```diff
@@ -33,81 +35,117 @@ const Profile = ({ userId, booksReadCount }) => {
+  }, []);
+
+  // Memoize updateBooksReadCount to prevent unnecessary re-renders
+  const updateBooksReadCount = useCallback(
```
I don't quite see where is that used, but this could be because of the github diff. From what I see though, the function only uses a state setter inside so it does not need to be memoized - the maintenance of that is handled by the library

tanya-newStart committed:
Wrap app into BookshelfProvider

tanya-newStart committed:
Adjust book context to include fetching books

tanya-newStart committed:
Make sure bookShelf is an object

tanya-newStart committed:
Use bookshelf context in Profile

tanya-newStart committed:
Add bookshelf context to Bookshelf

tanya-newStart committed:
Use useAuth() in FavoriteQuote

tanya-newStart committed:
Add useCallback in Bookshelf

pankaj28843 committed:
fix: use server-returned book data to ensure unique book_id

- Changed add book logic to use book data returned from server instead of request payload. This will ensure the list where the new book is rendered always has a unique book_id, thus avoiding React errors related to non-unique keys on list children.

magdazelena reviewed (commented):
👏
